### PR TITLE
Ensure topk results for IVF_FLAT_CC

### PR DESCRIPTION
**Pull request by chasingegg**
#354 
/kind improvement

### CODE REVIEW

File: src/index/ivf/ivf.cc [hunk 1]
```diff
@@ -541,17 +541,26 @@ IvfIndexNode<DataType, IndexType>::Search(const DataSet& dataset, const Config&
                             distances[i + offset] = static_cast<float>(i_distances[i + offset]);
                         }
                     }
-                } else if constexpr (std::is_same<IndexType, faiss::IndexIVFFlat>::value) {
+                } else if constexpr (std::is_same<IndexType, faiss::IndexIVFFlatCC>::value) {
```
Comment (alexanderguzhva) [inline src/index/ivf/ivf.cc:544]: We're not using `IndexIVFFlat`, just `IndexIVFFlatCC` all the time, right?

File: thirdparty/faiss/faiss/IndexIVF.h [hunk 2]
```diff
@@ -71,6 +71,7 @@ struct Level1Quantizer {
 struct SearchParametersIVF : SearchParameters {
     size_t nprobe = 1;    ///< number of probes at query time
     size_t max_codes = 0; ///< max nb of codes to visit to do a query
+    bool ensure_topk_full = false; ///< indicate whether we make sure topk result is full
```
Comment (alexanderguzhva) [inline thirdparty/faiss/faiss/IndexIVF.h:74]: I'd add more comments here. It did not become clear what this parameter does before I read the code :)

File: thirdparty/faiss/faiss/IndexIVF.h [hunk 3]
```diff
@@ -493,7 +494,8 @@ struct InvertedListScanner {
             const idx_t* ids,
             float* distances,
             idx_t* labels,
-            size_t k) const;
+            size_t k,
+            size_t& scan_cnt) const;
```
Comment (alexanderguzhva) [inline thirdparty/faiss/faiss/IndexIVF.h:498]: Please add comments to the description of the function.

**Comment by alexanderguzhva** [comment]
/lgtm
This change will need to be merged wirh the baseline whenever time permits. I'll do that.

File: src/index/ivf/ivf.cc (hunk 1 above)
Comment (chasingegg) [reply src/index/ivf/ivf.cc:544]: Actually I dont know why we need a seperate code branch for IndexIVFFlat, it is the same with IVFPQ, IVFSQ, etc

File: thirdparty/faiss/faiss/IndexIVF.h (hunk 2 above)
Comment (chasingegg) [reply thirdparty/faiss/faiss/IndexIVF.h:74]: Sure, I will add more comments in the next pr.

**Comment by chasingegg** [comment]
> /lgtm This change will need to be merged wirh the baseline whenever time permits. I'll do that.

Yeah, I've opened an issue at faiss repo, it is actually a break change to current faiss implementation on 'scan_count' meaning, and I somehow prefer the meaning in knowhere :)

File: src/index/ivf/ivf.cc (hunk 1 above)
Comment (chasingegg) [reply src/index/ivf/ivf.cc:544]: But we do need a new code branch for IndexIVFFlatCC, so I replace it.

**Pull request merged**